

PACKAGE = "lidar_inten_detect"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("shelf_width", double_t, 0, "Shelf width", 0.60, 0.1, 2.0)
gen.add("shelf_width_tolerance", double_t, 0, "Shelf width tolerance", 0.10, 0.01, 0.5)
gen.add("intensity_threshold", int_t, 0, "Intensity threshold", 35, 0, 100)
gen.add("eps", double_t, 0, "DBSCAN eps", 0.25, 0.05, 1.0)
gen.add("min_samples", int_t, 0, "DBSCAN min samples", 10, 1, 50)
gen.add('dist_threshold', double_t, 0, "Distance threshold (meters)", 0.5, 0.0, 5.0)
gen.add('offset_length', double_t, 0, "Offset length for back legs (meters)", 0.60, 0.0, 2.0)

exit(gen.generate(PACKAGE, "cfg", "DistanceCalculator"))

